### ISSUE TITLE
test_mplot3d:test_quiver3d broke unintentionally

### ISSUE THREAD

**Issue by mdboom** [matplotlib/matplotlib#5531]
In #5307, where I am lowering our testing tolerance down to zero, I am discovering a few recent changes that slipped through because the tolerance on our tests was too high.

This test was broken in 72819e9617b464236724ccf8d0cc2a02c20560bc.

BASELINE IMAGE:

![quiver3d-expected](https://cloud.githubusercontent.com/assets/38294/11304612/d405c35c-8f76-11e5-91cf-6bf21bab718a.png)

CURRENT RESULT:

![quiver3d](https://cloud.githubusercontent.com/assets/38294/11304618/dcec7eac-8f76-11e5-8ccb-f5f10253833a.png)

DIFFERENCE:

![quiver3d-failed-diff](https://cloud.githubusercontent.com/assets/38294/11304627/ebd97b90-8f76-11e5-863a-69f1bf33f5c9.png)

TEST CODE:

```
@image_comparison(baseline_images=['quiver3d'], remove_text=True)
def test_quiver3d():
    fig = plt.figure()
    ax = fig.gca(projection='3d')

    x, y, z = np.ogrid[-1:0.8:10j, -1:0.8:10j, -1:0.6:3j]

    u = np.sin(np.pi * x) * np.cos(np.pi * y) * np.cos(np.pi * z)
    v = -np.cos(np.pi * x) * np.sin(np.pi * y) * np.cos(np.pi * z)
    w = (np.sqrt(2.0 / 3.0) * np.cos(np.pi * x) * np.cos(np.pi * y) *
            np.sin(np.pi * z))

    ax.quiver(x, y, z, u, v, w, length=0.1, pivot='tip', normalize=True)
```

I don't *think* this change was intentional, so we need to figure out what in 72819e9617b464236724ccf8d0cc2a02c20560bc is unintentionally causing this change.

Cc: @perimosocordiae, @WeatherGod

**Comment by perimosocordiae** [comment]
Yep, that looks like a real bug. I'll look into it.

**Comment by WeatherGod** [comment]
The three whole arrows that we see going missing are all in critical points
in the divergence field. I bet that the vectorization work change the
behavior at those locations, maybe resulting in NaNs now?

**Comment by WeatherGod** [comment]
All of the other tiny differences is probably due to changes in numerical
precision of the calculation.

**Comment by mdboom** [comment]
There's another test that may be related (I haven't confirmed through git bisection, but thought I'd bring it to attention here in case it's helpful).  `test_quiver3d_mask`

BASELINE IMAGE:

![quiver3d_masked-expected](https://cloud.githubusercontent.com/assets/38294/11305096/ef8d2154-8f78-11e5-8c01-fe7e2aae551b.png)

CURRENT RESULT:

![quiver3d_masked](https://cloud.githubusercontent.com/assets/38294/11305102/f77f7682-8f78-11e5-95bb-d6ef2b4d6dcc.png)

DIFF:

![quiver3d_masked-failed-diff](https://cloud.githubusercontent.com/assets/38294/11305111/00d39de4-8f79-11e5-8039-8fcb2d6b2431.png)

**Comment by mdboom** [comment]
Yeah -- I'm not so much worried about the subpixel differences in numerical precision.  We can just update the images for that.  The missing arrows, as you say, are the real concern.

**Comment by perimosocordiae** [comment]
Found the bug. I'll open a PR with the fix shortly.

**Comment by mdboom** [comment]
This was included as part of #5307, now closed.
